design/account.go

```diff
@@ -141,7 +31,27 @@ var _ = a.Resource("user", func() {
 		)
 		a.Description("Get the authenticated user in JSON-API format")
 		a.UseTrait("conditional")
-		a.Response(d.OK, user)
+		a.Response(d.OK, showUser)
+		a.Response(d.NotModified)
+		a.Response(d.BadRequest, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+	})
+
+	a.Action("listResources", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET("/resources"),
+		)
+		a.Params(func() {
+			// requirement and value will be handled by the controller method,
+			// in order to be able to return a proper JSON-API response to the client in case
+			// of a bad request
+			a.Param("type", d.String, "the type of resource to list")
```

Comment:
How about adding sample values here? Currently, we are only supporting `space`. This way it will helpful to understand for consumer about what are supported types and they can use the same. Something like `Enum("spaces")`

Reply:
ok, let me see that

Reply:
actually, we could do that, but then Goa will immediately return an error while processing the request with an invalid or empty `type` parameter, without calling the controller, which prevents us from returning a JSON-API response. So we need to be permissive in the design and handle the `type` value in the controller.

Reply:
(and now I remember that I tried that but reverted the change as I could not test the case of invalid/empty type parameter as I needed to)